src/new-components/i18n-helpers.ts

```diff
@@ -0,0 +1,29 @@
+import { useTranslation } from 'gatsby-plugin-react-i18next';
+import { format, parseISO } from 'date-fns';
+import { enGB, de } from 'date-fns/locale';
+/**
+ * A custom  hook to read the current language
+ * and return a formatter (date-fns) with the correct locale set
+ * with the default being `enGB` for any unknown language.
+ */
+export const useLocalFormat = (dateFormat) => {
```

Comment:
First I had a format factory function with a lang parameter. I recognized that the react way is to use a custom hook which acts as the same factory anyway. feels good 👍

Reply:
Typo alarm 🚨 

`useLocalFormat` -> `useLocaleFormat`
`getLocal` -> `getLocale`

Reply:
thanks for the catch! fixed